examples/hello_compute_c/main.c

```diff
@@ -0,0 +1,139 @@
+#include "./../../ffi/wgpu.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+#define WGPU_TARGET_MACOS 1
+#define WGPU_TARGET_LINUX 2
+#define WGPU_TARGET_WINDOWS 3
+
+#define BLEND_STATES_LENGTH (1)
+#define ATTACHMENTS_LENGTH (1)
+#define RENDER_PASS_ATTACHMENTS_LENGTH (1)
+#define BIND_GROUP_LAYOUTS_LENGTH (1)
+
+WGPUByteArray read_file(const char *name) {
+    FILE *file = fopen(name, "rb");
+    fseek(file, 0, SEEK_END);
+    long length = ftell(file);
+    unsigned char *bytes = malloc(length);
+    fseek(file, 0, SEEK_SET);
+    fread(bytes, 1, length, file);
+    fclose(file);
+    return (WGPUByteArray){
+        .bytes = bytes,
+        .length = length,
+    };
+}
+
+void read_buffer_map(
+    WGPUBufferMapAsyncStatus status, const uint8_t *data, uint8_t *userdata) {
+    if (status == WGPUBufferMapAsyncStatus_Success) {
+        printf("Times: %s \n", data);
```

Comment:
- `data` here is a pointer (`uint8_t*`).
- `data` at this point will be pointing to the data in our staging buffer, which we know should have a size of 16 bytes.
- These 16 bytes should also be an array, containing four `uint32_t`s.
- If we know the size is correct (i.e. it's aligned for all elements of the array), we can safely cast it between types. For example, we want to read `uint32_t*` but it's currently a `uint8_t*`, so we can simply do `uint32_t *times = (uint32_t *)data;` to cast it.
- Now `times[0]` will contain the first `uint32_t` value in our staging buffer, `times[1]` will contain the second, then `times[2]`, then `times[3]`
- We can print this to match Rust formatting with something like `        printf("Times: [%d, %d, %d, %d]", times[0], times[1], times[2], times[3]);` or a `for` loop